ARCHITECTURE.md

```diff
@@ -6,6 +6,233 @@ This document describes the high-level architecture of Ditto Network JS SDK.
+  "triggers": [
+    {
+      "type": "Core.Scheduled",
+      "options": {
+        "frequency": "daily",
+        "time": "08:00"
+      }
+    }
+  ],
+  "actions": [
+    {
+      "type": "Application.MultiSender",
+      "options": {
+        "items": [
+          {
+            "asset": {
+              "address": "0xTokenAddress",
+              "decimals": 18
+            },
+            "amount": "100",
+            "to": "0xRecipientAddress"
+          },
+          {
+            "asset": null,
+            "amount": "0.1",
+            "to": "0xAnotherRecipientAddress"
+          }
+        ]
+      }
+    }
+  ]
+}
+```
+
+
+### Trigger and Action Objects
+
+#### Trigger
+
+A trigger object defines the conditions under which the workflow is executed.
+
+- **type**: (String) The type of trigger. This should correspond to a predefined trigger type in the application.
+- **options**: (Object) An optional set of parameters specific to the trigger type.
+
+```json
+{
+  "type": "Core.Scheduled",
+  "options": {
+    "frequency": "daily",
+    "time": "08:00"
+  }
+}
+```
+
+- **type**: Specifies a scheduled trigger.
+- **options**:
+  - **frequency**: The frequency with which the trigger should activate (e.g., "daily", "weekly").
+  - **time**: The specific time of day the trigger should activate.
+
+
+#### Action
+
+An action object defines the operations that are performed when the workflow is triggered.
+
+- **type**: (String) The type of action. This should correspond to a predefined action type in the application.
+- **options**: (Object) An optional set of parameters specific to the action type.
+
+
+```json
+{
+  "type": "Application.MultiSender",
+  "options": {
+    "items": [
+      {
+        "asset": {
+          "address": "0xTokenAddress",
+          "decimals": 18
+        },
+        "amount": "100",
+        "to": "0xRecipientAddress"
+      },
+      {
+        "asset": null,
+        "amount": "0.1",
+        "to": "0xAnotherRecipientAddress"
+      }
+    ]
+  }
+}
+```
+
+- **type**: Specifies a multi-send action by the application.
+- **options**:
+  - **items**: An array of objects representing the assets to be sent, each containing:
+    - **asset**: An object representing the ERC20 token, or `null` for native tokens.
+      - **address**: The address of the ERC20 token.
+      - **decimals**: The number of decimals for the ERC20 token.
+    - **amount**: The amount of the asset to send.
+    - **to**: The recipient address.
+
+
+### Extending Configuration
+
+This format is designed to be extensible. Additional triggers and actions can be defined by specifying new types and corresponding options. This approach allows for a highly customizable and adaptable workflow configuration system.
+
+```ts
+const wf = workflowFactory
+  // Registers an action with the workflow using its type and the corresponding options
+  .registerAction('Application.MultiSender', MultiSenderAction) // => new MultiSenderAction(options)
```

Comment:
Why do we need to duplicate type when registering? Can't we just pull it from action config?

Reply:
Yes. Good point, it could public property of Action, like `MultiSenderAction.id`. What do you think?